lib/rides/pages/ride_detail_page.dart

```diff
@@ -154,13 +159,33 @@ class _RideDetailPageState extends State<RideDetailPage> {
     return Scaffold(
       appBar: AppBar(
         title: Text(S.of(context).pageRideDetailTitle),
-        actions: <Widget>[
-          IconButton(
-            onPressed: () {},
-            icon: const Icon(Icons.chat),
-            tooltip: S.of(context).openChat,
-          )
-        ],
+        actions: _fullyLoaded
+            ? <Widget>[
+                IconButton(
+                  onPressed: () => Navigator.push(
+                    context,
+                    MaterialPageRoute<void>(
+                      builder: (BuildContext context) => ChatPage(
+                        chatId: _ride!.chat?.id,
+                        profile: _ride!.drive!.driver!,
+                        active: _ride!.status.activeChat(),
+                      ),
+                    ),
+                  ),
+                  icon: Badge(
+                    badgeContent: Text(
```

Comment:
the badge does not work because you don't load the messages in the chat query. that's why I did not want the
"if(messages == null) return", because it covers up these kinds of mistakes